RandLAPACK/comps/rl_util.hh

```diff
@@ -552,5 +813,47 @@ int64_t rank_search_binary(
     return k;
 }
 
+/// Iterates the upper-triangular matrix A from end to beginning and 
+/// attempts to estimate smallest k such taht ||A[k:, k:]||_F <= tau_trunk * ||A||_2.
+/// Computes matrix Fro norms through vector norm updates.
+template <typename T>
+int64_t rank_search_linear(
+    int64_t n,
+    T norm_2_A,
+    T tau_trunc,
+    T* A_dat
+) {
+    T norm_A_work = 0.0;
+    T norm_A_row = 0.0;
+    for(int i = n - 1, j = 1; i > 0; --i, ++j) {
+        norm_A_row = blas::nrm2(j, A_dat + ((n + 1) * i), n);
+        // Add norm stably
+        norm_A_work = std::hypot(norm_A_work, norm_A_row);
+
+        if(norm_A_work > tau_trunc * norm_2_A)
+            return i + 1;
+    }
+    return 1;
```

Comment:
What happens when this function is applied to the zero matrix?

Reply:
This function has a very precise use case and will not be applied to a zero matrix.

Reply:
Either add that requirement to the documentation or include the special casing to catch the zero matrix. Either is fine with me.

Reply:
Also, i kind of feel like we can get rid of linear search. It inly complicates things.

Reply:
I'm fine with removing linear search

Reply:
@TeachRaccooon please make sure binary search works with the zero matrix.

Reply:
Done.